robotoff/insights/importer.py

```diff
@@ -1043,6 +1043,36 @@ def generate_candidates(
             yield insight
 
 
+class UPCImageImporter(InsightImporter):
+    """
+    Insight importer for UPC images
+    """
+
+    @staticmethod
+    def get_type() -> InsightType:
+        return InsightType.is_upc_image
+
+    @staticmethod
+    def get_required_prediction_types() -> set[PredictionType]:
+        return {PredictionType.is_upc_image}
+
+    @classmethod
+    def is_conflicting_insight(
+        cls, candidate: ProductInsight, reference: ProductInsight
+    ) -> bool:
+        return candidate.source_image == reference.source_image
+
+    @classmethod
+    def generate_candidates(
+        cls,
+        product: Optional[Product],
+        predictions: list[Prediction],
+    ) -> Iterator[ProductInsight]:
+        for p in predictions:
```

Comment:
We should check here using `product` that the `source_image` refers to a select image (same check as in annotator): it's useless to create insights that cannot be applied.

Reply:
Not 100% sure on how to go about this. Would it look like the following:

```
for p in predictions:
            image_id = get_image_id(p.source_image or "")
            if not product or image_id is None or image_id not in product["images"]:
                logger.info(
                    "Image %s not found for product %s, failed generating insight",
                    image_id or "unknown",
                    product.barcode or "unknown",
                )
                continue
            
            insight = ProductInsight(**p.to_dict())
            yield insight```